tests/test_codecs.py

```diff
@@ -452,6 +452,20 @@ def test_bitstring(self):
         res = await self.con.fetchval("SELECT '-5 years -1 month'::interval")
         self.assertEqual(res, datetime.timedelta(days=-1855))
 
+    async def test_numeric(self):
+        res = await self.con.fetchval(
+            "SELECT $1::numeric", decimal.Decimal('NaN'))
+        self.assertTrue(res.is_nan())
+
+        with self.assertRaisesRegex(ValueError, 'numeric type does not '
```

Comment:
Python also supports signalling NaN (`sNaN`) -- add a check that we handle it with a proper exception.